polaris/competition/_base.py

```diff
@@ -0,0 +1,229 @@
+from polaris.experimental._dataset_v2 import DatasetV2
+from polaris.utils.dict2html import dict2html
+from polaris.utils.types import TargetType, TaskType, SplitType
+
+ColumnsType: TypeAlias = str | list[str]
+
+
+class CompetitionSpecification(DatasetV2):
+    """An instance of this class represents a Polaris competition. It defines fields and functionality
+    that in combination with the [`DatasetV2`][polaris.experimental._dataset_v2.DatasetV2] class, allow
+    users to participate in competitions hosted on Polaris Hub.
+
+    Examples:
+        Basic API usage:
+        ```python
+        import polaris as po
+
+        # Load the benchmark from the Hub
+        competition = po.load_competition("polaris/hello-world-competition")
+
+        # Get the train and test data-loaders
+        train, test = competition.get_train_test_split()
+
+        # Use the training data to train your model
+        # Get the input as an array with 'train.inputs' and 'train.targets'
+        # Or simply iterate over the train object.
+        for x, y in train:
+            ...
+
+        # Work your magic to accurately predict the test set
+        prediction_values = np.array([0.0 for x in test])
+        predictions = CompetitionPredictions(
+            name="first-prediction",
+            owner=HubOwner(slug="dummy-user"),
+            report_url="REPORT_URL",
+            target_labels=competition.target_cols,
+            test_set_labels=competition.test_set_labels,
+            test_set_sizes=competition.test_set_sizes,
+            predictions=prediction_values,
+        )
+
+        # Submit your predictions
+        competition.submit_predictions(predictions)
+        ```
+
+    Attributes:
+        target_cols: The column(s) of the original dataset that should be used as target.
+        input_cols: The column(s) of the original dataset that should be used as input.
+        split: The predefined train-test split to use for evaluation.
+        metrics: The metrics to use for evaluating performance
+        main_metric: The main metric used to rank methods.
+        readme: Markdown text that contains a formatted description of the competition.
+        target_types: A dictionary that maps target columns to their type.
+        start_time: The time at which the competition starts accepting prediction submissions.
+        end_time: The time at which the competition stops accepting prediction submissions.
+        n_test_sets: The number of test sets defined in the test split.
+        n_test_datapoints: The number of datapoints in each test set.
+        n_classes: The number of classes within target columns that define a classification task.
+    For additional meta-data attributes, see the [`DatasetV2`][polaris.experimental._dataset_v2.DatasetV2] class.
+    """
+
+    _artifact_type = "competition"
+
+    target_cols: ColumnsType
+    input_cols: ColumnsType
+    split: SplitType
```

Comment:
**Fix:** You'll need a serializer and validator for the split!

Reply:
For my understanding, why are they needed?

Serializer: If we define each competition specification ourselves, and we choose to use a python list (as we usually do), why is a serializer needed?

Validator: If we define each competition ourselves, can't we assume that the split is valid on retrieval?

I'm not against adding them, I just assumed that we were ok with less validation because we define/upload the competitions on our side.

Reply:
Even if it's just us uploading, the data models are quite complex and the validators have helped me spot errors a few times already. Especially since we have the code for the validation already, I think it will make our lives easier!